### PR TITLE
Add VGmanager daemonset, VGManager implementation

### PR DESCRIPTION
**Pull request by rohantmp**
Will be functional today, will comment on the PR when it is. Basic structure is in place.

Pseudocode comments are in place and some functions are empty.

TODO after PR posted:

- [x] fix CI
- [x] copy blockutil from 
- [x] write lvmd wrapper
- [x] implement `filterAvailableDevices`
- [x] implement  `addMatchingDevicesToVG` (theoretically functional at this point)
- [x] double check error handling
- [ ] write unit tests
- [ ] report daemonset health in status

### CODE REVIEW

**Comment by rohantmp** [comment]
I think it's ready for review, but I can't test latest changes because of build issues.

**Review by leelavg — COMMENTED** [review]
- didn't review `internal` and some of the pkgs in `vgmanager`

File: controllers/vgmanager.go [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+				NodeAffinity: &corev1.NodeAffinity{
+					RequiredDuringSchedulingIgnoredDuringExecution: nodeSelector,
+				},
+			}
+		} else {
+			ds.Spec.Template.Spec.Affinity = nil
+		}
+		return nil
+	})
+	if result != controllerutil.OperationResultNone {
```
[the remark this answers is not in the document]
Comment (rohantmp) [reply controllers/vgmanager.go]: this logs everything other than operationresultnone

**Comment by rohantmp** [comment]
That should do it. Lacks polish and unit-testing, but I've run out of time.

 I've left todo(rohan) comments wherever I couldn't get to things in time.

Here's a demo of things in action: https://asciinema.org/a/nmhZQYXgTLG3oSOxE0ROD8vlX

File: Dockerfile.vgmanager [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+COPY api/ api/
+COPY pkg/ pkg/
+COPY controllers/ controllers/
+
+# Build
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a -o vgmanager cmd/vgmanager/main.go
+
+# Use distroless as minimal base image to package the manager binary
+# Refer to https://github.com/GoogleContainerTools/distroless for more details
+FROM gcr.io/distroless/static:nonroot
```
[the remark this answers is not in the document]
Comment (rohantmp) [reply Dockerfile.vgmanager]: switched to centos

**Review by nbalacha — CHANGES_REQUESTED** [review]

**Comment by leelavg** [comment]
- Parts of this PR #26 and #27 is merged and so closing this.

**Pull request closed without merging**